savant/deepstream/pipeline.py

```diff
@@ -121,6 +132,67 @@ def __init__(
 
         super().__init__(name, pipeline_cfg, **kwargs)
 
+    def _add_queues_to_pipeline(
+        self,
+        pipeline_cfg: Pipeline,
+        buffer_queues: BufferQueuesParameters,
+    ):
+        """Add queues to the pipeline before and after pyfunc elements."""
+
+        queue_properties = {
+            'max-size-buffers': buffer_queues.length,
+            'max-size-bytes': buffer_queues.byte_size,
+            'max-size-time': 0,
+        }
+        self._add_queues_to_element_group(
+            element_group=pipeline_cfg,
+            queue_properties=queue_properties,
+            last_is_queue=False,
+            next_should_be_queue=False,
+        )
+
+    def _add_queues_to_element_group(
+        self,
+        element_group: Union[Pipeline, ElementGroup],
+        queue_properties: Dict[str, Any],
+        last_is_queue: bool,
+        next_should_be_queue: bool,
+    ):
+        """Add queues to the pipeline or an element group before and after pyfunc elements."""
+
+        elements = []
+        for i, element in enumerate(element_group.elements):
+            if isinstance(element, ElementGroup):
+                if not element.init_condition.is_enabled:
+                    continue
+
+                last_is_queue = self._add_queues_to_element_group(
+                    element,
+                    queue_properties,
+                    last_is_queue,
+                    next_should_be_queue,
+                )
+                next_should_be_queue = False
+                elements.append(element)
+                continue
+
+            if (next_should_be_queue and element.element != 'queue') or (
```

Comment:
I think you can also check if there are elements before/after pyfunc. You don't need to add a queue between muxer and pyfunc and pyfunc and demuxer.

Reply:
Fixed it.